app/src/main/java/com/hedvig/app/util/LCE.kt

```diff
@@ -0,0 +1,7 @@
+package com.hedvig.app.util
+
+sealed class LCE<out T> {
```

Comment:
Cool! Do you think we can use this in other parts of the app? 

Is "LCE" a name from the functional world, like Left/Right?

Reply:
Hmm yeah I haven't decided 100% on this either.

LCE is just short for Loading/Content/Error and all it does in this case is save us from creating yet another model in the case where whatever we're doing is adequately modeled with these three states.
Technically if anything else in the app is similarly modeled with the exact same 3 possible outcomes, they could use this LCE in order to just save time and the mental and IDE overhead of having to import yet another type which just contains those 3 cases inside of it, with its own package name etc which makes auto-completion harder.
Example of something we have in multiple places in the app:
![image](https://user-images.githubusercontent.com/44558292/145576864-d92d7bb8-86d1-43ec-b7ca-1c1744d3ec1e.png)

Technically the same can be done for a result type where we can migrate all of our custom Result objects into [Insert whatever library here].
But all in all, this mostly saves on an extra type that has to live inside the class we're working on. 
![image](https://user-images.githubusercontent.com/44558292/145576315-31edc56b-3343-4419-945c-dd192ab3c82b.png)

In the end, whenever any of the cases broke out of the "Can be represented by these 2-3 standard cases" then we could go custom as we're doing already. That's why I'm not sure on it, I am afraid if we do it wrong it will just arbitrary constraints that don't need to be there in the first place.
But yeah I don't know, I don't have a strong opinion on it as of now, just trying out things and see what makes the most sense and we can decide on something all together if we feel like doing so.

Reply:
I like it - and have thought of something similar. For this PR though I think we can go with what we currently have (eg have it in the viewmodel). Lets continue the discussion in slack!

Reply:
Reverted for now as suggested. We can always come back to this in the future if we have more ideas! 😊